lib/url-decorator.jsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { CompositeDecorator } from 'draft-js';
+import urlRegex from 'url-regex';
+import normalizeUrl from 'normalize-url';
+import { viewExternalUrl } from './utils/url-utils';
+
+const urlRe = urlRegex();
+/*
+- markdownLinkRe is a regexp to be applied to the substring preceding a matched url, ie. `wholeString.slice(0, wholeString.indexOf(matchedUrl))`
+- it tests whether the matched url is preceded by the charcters that define a link in markdown, ie. `(` or `]: `
+*/
+const markdownLinkRe = /(?:\(|\]:\s+)$/;
+
+export function normalizeUrlsInMarkdownString( str ) {
+	return str.replace( urlRe, ( match, offset ) => {
+		var normalized = normalizeUrl( match );
+		if ( markdownLinkRe.test( str.slice( 0, offset ) ) ) {
+			return normalized;
+		}
+		// replace url with markdown link only if it's not already part of one
+		return `[${match}](${normalized})`;
+	} );
+}
```

Comment:
the only reason I did this to linkify everything is because it was kind of weird to see some urls highlighted in edit mode, switch to preview mode and now they're not urls anymore.
but you do have a point that this is no longer markdown (it's probably one of the myriad flavours of markdown out there).
what do you think is more important *consistency between edit and preview mode* or *sticking to pure markdown output*?
-